live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/transform/response/TransformResponse.java

```diff
@@ -23,22 +23,19 @@
  * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
  * #L%
  */
-
 package org.alfresco.hxi_connector.live_ingester.adapters.messaging.transform.response;
 
 import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
 
-import lombok.AllArgsConstructor;
-import lombok.Getter;
-import lombok.NoArgsConstructor;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import org.springframework.validation.annotation.Validated;
 
-@Getter
+import org.alfresco.hxi_connector.live_ingester.adapters.messaging.transform.model.ClientData;
+import org.alfresco.hxi_connector.live_ingester.adapters.messaging.transform.model.ClientDataDeserializer;
+
 @Validated
-@NoArgsConstructor
-@AllArgsConstructor
-public class TransformResponse
-{
-    @NotBlank
-    private String targetReference;
-}
+public record TransformResponse(
+        @NotBlank String targetReference,
+        @NotNull @JsonDeserialize(using = ClientDataDeserializer.class) ClientData clientData)
```

Comment:
I think you can add this deserializer in `MappingConfig` instead

Reply:
Annotations are more descriptive IMO and tell straightforwardly that something is handled differently, in the other hand I don't mind hiding this in `MappingConfig` and simplifying record 🤔

Reply:
It turned out, that when I register the deserilizer globally then the `RawJsonDeserializer` which uses global `ObjectMapper` calls itself again in some kind of circular relation but without the JSON value and fails. I will leave it as it is and we can change it in the future.